ballista/rust/core/src/client.rs

```diff
@@ -135,24 +135,28 @@ impl BallistaClient {
 }
 
 struct FlightDataStream {
-    stream: Streaming<FlightData>,
+    stream: Mutex<Streaming<FlightData>>,
```

Comment:
this is needed because after tonic/prost upgrade `FlightData` is no longer `Sync`. There are perhaps better ways to handle this